dogma.go

```diff
@@ -215,29 +215,28 @@ func (c *Client) do(ctx context.Context, req *http.Request, resContent interface
 
 	res, err := c.client.Do(req)
 	if err != nil {
-		select {
-		case <-ctx.Done():
-			return nil, ctx.Err()
-		default:
-		}
-
 		return nil, err
 	}
-	defer res.Body.Close()
+	defer func() {
+		// drain up 512 bytes and close the body to reuse connection
+		// see also:
+		// - https://github.com/google/go-github/pull/317
+		// - https://forum.golangbridge.org/t/do-i-need-to-read-the-body-before-close-it/5594/4
+		io.CopyN(ioutil.Discard, res.Body, 512)
```

Comment:
https://github.com/google/go-github/blob/master/github/github.go#L501
The current version deprecates this feature. Could you recheck it, please?

Reply:
I did a test on this when building some production and confirm that it still remains.

Reply:
Could you also change this as well please? :-)
https://github.com/google/go-github/pull/847/

Reply:
you mean add more refence link to comment @minwoox

Reply:
Ah I mean that we do not have to drain up 512 bytes since it was introduced in 1.7. So we can revert this to as `defer res.Body.Close()`.
Please check out https://github.com/google/go-github/issues/843
and https://github.com/golang/go/commit/18072adbca7ee2cb2019704577a8f3f142010ba7
:-)

Reply:
My team used 1.10.3 for some productions and without draining up, the connection could not be reused. 
The case is: server response some data but client only care about status code.

Reply:
PTAL @minwoox : https://gist.github.com/linxGnu/0511f18c768cbe8f3dc0061cd6a0b5fb
The result on my laptop:
```
go1.11
With drainup: 2.645
Without drainup: 42.728
```

Reply:
Yeah, but in our case, we read all the bytes when the status is Error and call `json.NewDecoder(res.Body)` when the status 2xx. So we consumed all the bytes, or no? I'm just worrying that the codes in the reference you mentioned are gone now and people might get confused when looking at the codes.

Reply:
@minwoox Year, currently we are doing json decoder. 
- in case server response with additional malform data, then decoder won't help us.
- the other case is unknown status which we refuse to consume or processing.
Both of case seem not happened in real production being since server always return well format data. However, because we are doing client, we should just ignore what server response and handle all malform things. WDYT @minwoox

Reply:
I agree with you on that we have to deal with the case when the response data is malformed.
I have thought about for a while why `google/go-github` removes the codes introduced. I believe that
- the client is only sending the request to the Github which emits the controled responses by them.
- they will get to know easily when the response is modified if they do not have this trick by looking at the new connection count so that they can notice the response is changed and fix it.
- they just want to make it simple by removing the code because there's no problem when the condition meets above situation.

Reply:
Perhaps we do not have useful metrics on Central Dogma, we'd better have this trick.

Reply:
@trustin, @hyangtack You guys have any opinion on this?

Reply:
OK with keeping this as long as it's well commented.